client.go

```diff
@@ -122,61 +144,85 @@ func (c *Client) FetchKeyQuota(ctx context.Context, accessKey, origin string, no
 // FetchUsage fetches the current usage of the access key.
 func (c *Client) FetchUsage(ctx context.Context, quota *proto.AccessQuota, now time.Time) (int64, error) {
 	key := getQuotaKey(quota.AccessKey.ProjectID, quota.Cycle, now)
+
+	logger := c.logger.With(
+		slog.String("op", "fetch_usage"),
+		slog.Uint64("project_id", quota.AccessKey.ProjectID),
+		slog.String("access_key", quota.AccessKey.AccessKey),
+	)
+
 	for i := time.Duration(0); i < 3; i++ {
```

Comment:
How the loop is intended to work?
If i am correct this is loop will run only 3 nano seconds.

I suggest to use this pattern:
```go
	start := time.Now()
	duration := 3 * time.Second

	for time.Since(start) < duration {
	}
```

Or if you just prepare the int with type `time.Duration` than to that in code itself please. It can be confusing, that i think it is doing something more, then 3 loops.
```go
for i := 0; i < 3; i++ {
	...
		time.Sleep(time.Millisecond * 100 * (time.Duration(i) + 1))
	...
}
```
Or since go version 1.22 we can do also this, but i still didn't decided if i like or not :D
```go
for i := range 3 {
	...
		time.Sleep(time.Millisecond * 100 * (time.Duration(i) + 1))
	...
}